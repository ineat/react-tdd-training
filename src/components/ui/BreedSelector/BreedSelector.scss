.breed-selector {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;

  .breed {
    border: 3px solid lightslategrey;
    padding: 1rem;
    border-radius: 25px;
    margin: 0.5rem;
    font-size: 20px;
    transition: background-color 0.5s;

    &:hover {
      cursor:pointer;
      background-color: burlywood;
    }
  }
}