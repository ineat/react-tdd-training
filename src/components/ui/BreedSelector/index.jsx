import * as React from 'react';

import './BreedSelector.scss';

export default ({ breeds, onSelectBreed }) => (
  <div className="breed-selector">
    {
      breeds.map(breed => (
        <div className="breed" key={ breed.id } onClick={() => onSelectBreed(breed.id)}>
          { breed.name }
        </div>
      ))
    }
  </div>
);