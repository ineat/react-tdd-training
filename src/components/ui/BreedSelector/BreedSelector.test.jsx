import * as React from 'react';
import { shallow, mount } from 'enzyme';

import BreedSelector from '.';

describe('Breed selector', () => {
  it('Should render.', () => {
    const props = {
      breeds: [
        {
          id: 'id-99',
          name: 'Rex'
        }, {
          id: 'id-66',
          name: 'Arg'
        }
      ]
    };

    expect(mount(<BreedSelector breeds={props.breeds} />)).toMatchSnapshot();
  });

  it('Should display as many breeds as given in args.', () => {
    const props = {
      breeds: [
        {
          id: 'id-1',
          name: 'Border collie'
        }, {
          id: 'id-2',
          name: 'Labrador'
        }, {
          id: 'id-3',
          name: 'Cocker'
        }, {
          id: 'id-4',
          name: 'Jack Russel'
        }
      ],
      onSelectBreed: jest.fn()
    };
    const wrapper = shallow(<BreedSelector breeds={props.breeds} />);

    const breeds = wrapper.find('.breed');
    expect(breeds).toHaveLength(4);
    expect(breeds.get(0).props.children).toEqual(props.breeds[0].name);
    expect(breeds.get(1).props.children).toEqual(props.breeds[1].name);
    expect(breeds.get(2).props.children).toEqual(props.breeds[2].name);
    expect(breeds.get(3).props.children).toEqual(props.breeds[3].name);
  });

  it('Should display as many breeds with name as given in another args.', () => {
    const otherProps = {
      breeds: [
        {
          id: 'id-5',
          name: 'Berger Allemand'
        }, {
          id: 'id-6',
          name: 'Berger Australien'
        }, {
          id: 'id-8',
          name: 'Didier'
        }
      ]
    };
    const wrapper = shallow(<BreedSelector breeds={otherProps.breeds} />);

    const breeds = wrapper.find('.breed');
    expect(breeds).toHaveLength(3);
    expect(breeds.get(0).props.children).toEqual(otherProps.breeds[0].name);
    expect(breeds.get(1).props.children).toEqual(otherProps.breeds[1].name);
    expect(breeds.get(2).props.children).toEqual(otherProps.breeds[2].name);
  });

  it('Should call "onSelectBreed" prop with breed id when clicking on breed id.', () => {
    const props = {
      breeds: [
        {
          id: 'id-1',
          label: 'Border collie'
        }
      ],
      onSelectBreed: jest.fn()
    };
    const wrapper = shallow(<BreedSelector {...props} />);

    const onSelectBreed = wrapper.find('.breed').prop('onClick');

    onSelectBreed();

    expect(props.onSelectBreed).toHaveBeenCalledWith(props.breeds[0].id);
  });
});