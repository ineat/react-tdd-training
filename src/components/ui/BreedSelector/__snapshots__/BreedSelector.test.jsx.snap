// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breed selector Should render. 1`] = `
<_default
  breeds={
    Array [
      Object {
        "id": "id-99",
        "name": "Rex",
      },
      Object {
        "id": "id-66",
        "name": "Arg",
      },
    ]
  }
>
  <div
    className="breed-selector"
  >
    <div
      className="breed"
      key="id-99"
      onClick={[Function]}
    >
      Rex
    </div>
    <div
      className="breed"
      key="id-66"
      onClick={[Function]}
    >
      Arg
    </div>
  </div>
</_default>
`;
